Compilation error no matching constructor for initialization of 'pythonic::types::contiguous_slice' (aka 'cstride_slice<1>')

```
#pythran export main()
import numpy as np

N_CONST = 5

def do_it(A, X):
    A[1][0][1][0] += 1
    return np.argsort(-X)[0:A[0][0]]

def main():
    N = N_CONST
    B = tuple(([np.empty((1, N))], [0]))
    A = tuple(([N], [B]))
    do_it(A, np.zeros(10))
    return 0
```
The code above generates this compilation error:
```
/var/folders/2p/dngs2w5x6tn3lw2jqsnl42_80000gp/T/tmpx3pw5r_u.cpp:94:106: error: no matching constructor for initialization of 'pythonic::types::contiguous_slice' (aka 'cstride_slice<1>')
      return pythonic::types::as_const(pythonic::numpy::functor::argsort{}(pythonic::operator_::neg(X)))[pythonic::types::contiguous_slice(0L,std::get<0>(pythonic::types::as_const(std::get<0>(pythonic::types::as_const(A)))))];
                                                                                                         ^                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/assignable.hpp:45:40: note: in instantiation of function template specialization '(anonymous namespace)::__pythran_bug_fail::do_it::operator()<std::tuple<(anonymous namespace)::pythonic::types::list<std::integral_constant<long, 5>>, (anonymous namespace)::pythonic::types::list<std::tuple<(anonymous namespace)::pythonic::types::list<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 1>, std::integral_constant<long, 5>>>>, (anonymous namespace)::pythonic::types::list<long>>>>, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 10>>>>' requested here
    return std::forward<F>(f).template operator()<by_val_t<Args>...>(
                                       ^
/var/folders/2p/dngs2w5x6tn3lw2jqsnl42_80000gp/T/tmpx3pw5r_u.cpp:143:24: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::call<(anonymous namespace)::__pythran_bug_fail::do_it, std::tuple<(anonymous namespace)::pythonic::types::list<std::integral_constant<long, 5>>, (anonymous namespace)::pythonic::types::list<std::tuple<(anonymous namespace)::pythonic::types::list<(anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 1>, std::integral_constant<long, 5>>>>, (anonymous namespace)::pythonic::types::list<long>>>>, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<std::integral_constant<long, 10>>>>' requested here
      pythonic::types::call(do_it(), (__type34)pythonic::builtins::functor::tuple{}(pythonic::types::make_tuple(__type12(std::integral_constant<long, 5>{}, pythonic::types::single_value()), __type19(B, pythonic::types::single_value()))), pythonic::numpy::functor::array{}(pythonic::types::make_tuple(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0), pythonic::numpy::functor::float64{}));
                       ^
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/slice.hpp:143:5: note: candidate constructor not viable: no known conversion from 'value_type' (aka 'std::integral_constant<long, 5>') to 'none<long>' for 2nd argument
    cstride_slice(none<long> lower, none<long> upper);
    ^
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/slice.hpp:137:10: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided
  struct cstride_slice {
         ^
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/slice.hpp:137:10: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 2 were provided
/Users/jlaroche/ua_conda_env-Darwin.arm64/lib/python3.13/site-packages/pythran/pythonic/include/types/slice.hpp:144:5: note: candidate constructor not viable: requires 0 arguments, but 2 were provided
    cstride_slice();
```

--- a/pythran/pythonic/include/types/NoneType.hpp
+++ b/pythran/pythonic/include/types/NoneType.hpp
@@ -116,6 +116,8 @@ namespace types
     none();
     none(none_type const &);
     none(T const &data);
+    template<class U, U V, class = std::enable_if<std::is_integral_v<U>>>
+    none(std::integral_constant<U, V> const &data) : none(T(data)) {}
     bool operator==(none_type const &) const;
     template <class O>
     bool operator==(O const &t) const;

--- a/pythran/tests/test_slice.py
+++ b/pythran/tests/test_slice.py
@@ -824,3 +824,17 @@ def slice_transpose0(n):
     slice1 = base[:10, 10:] # should have shape (10, 6)
     return slice1'''
         self.run_test(code, 16, slice_transpose0=[int])
+
+    def test_slice_bound_integral_constant(self):
+        code = '''
+import numpy as np
+N_CONST = 5
+def slice_bound_integral_constant(probs):
+    N = N_CONST
+    sub = np.empty((1, N))
+    # type inference flags state[0][0] as an integral constant
+    # we must be sure that a slice is constructible from it
+    state = ([N], sub)
+    return np.argsort(-probs)[0:state[0][0]]'''
+        self.run_test(code, numpy.zeros(16, dtype=numpy.float32),
+                      slice_bound_integral_constant=[NDArray[numpy.float32, :]])

